File: 3455-minimum-length-of-string-after-operations/3455-minimum-length-of-string-after-operations.c
int minimumLength (char* s) {
    int n = strlen (s), removableLen = 0;
    int freq [26] = {0};
    for (int i = 0; i < n; i++) {
        freq[s[i] - 'a']++;
    }

    for (int i = 0; i < 26; i++) {
        int frequency = freq [i];
        if (frequency % 2 != 0)
            removableLen += frequency - 1;
        else if (frequency != 0)
            removableLen += frequency - 2;
    }

    return n - removableLen;
}

File: 3455-minimum-length-of-string-after-operations/3455-minimum-length-of-string-after-operations.py
class Solution:
    def minimumLength (self, s : str) -> int:
        char_freq = [0] * 26
        total_length = 0
        for char in s:
            char_freq [ord (char) - ord ('a')] += 1
        for freq in char_freq:
            if freq == 0:
                continue
            elif freq % 2 == 0:
                total_length += 2
            else:
                total_length += 1
        return total_length